src/actions/blockchain.ts

```diff
@@ -49,13 +55,90 @@ export const setConnectionStatus = createAction<{ connected?: boolean }>('SET_CO
 export const setActiveProvider = createAction<{ provider?: string }>('SET_ACTIVE_PROVIDER')
 export const registerProvider = createAction<{ provider?: string, data?: Object }>('REGISTER_PROVIDER')
 export const updateProvider = createAction<{ provider?: string, data?: Object }>('UPDATE_PROVIDER')
-export const setCurrentBalance = createAction<{ provider?: string, currentBalance?: Balance }>('SET_CURRENT_BALANCE')
+export const setCurrentBalance = createAction<{ provider?: string, currentBalance?: Balance | BigNumber }>('SET_CURRENT_BALANCE')
 export const setCurrentAccountAddress =
   createAction<{ provider?: string, currentAccount?: Object }>('SET_CURRENT_ACCOUNT_ADDRESS')
 export const fetchTokens = createAction<{ tokens?: TokenBalances }>('FETCH_TOKENS')
 
+export const setFeeRatio = createAction<{ feeRatio: number }>('SET_FEE_RATIO')
+export const setTokenSupply = createAction<{ mgnSupply: string | BigNumber }>('SET_TOKEN_SUPPLY')
+
 const NETWORK_TIMEOUT = process.env.NODE_ENV === 'production' ? 10000 : 200000
 
+export const updateMainAppState = () => async (dispatch: Function) => {
+  const { TokenETH, TokenGNO, TokenOWL, TokenMGN } = await promisedContractsMap,
+    currentAccount = await getCurrentAccount()
+
+  // TODO: grab from IPFS defaultObj or uploaded file, or localStorage?
+  const defObj = [
```

Comment:
should we just add
```
MGN: 'MAGNOLIA',
OWL: 'OWL',
```
to `Code2Name` in `types/index.d.ts`?
then we can  do `const defObj: DefaultTokenList = [` and remove `//@ts-ignore` on lines 107 and 110

Reply:
Yeah, I like that